02-oop-inheritance/script.js

```diff
@@ -0,0 +1,93 @@
+class EventEmitter {
+  constructor() {
+    this.events = {};
+  }
+  on(eventName, callback) {
+    if (this.events[eventName]) {
+      this.events[eventName].push(callback);
+    } else {
+      this.events[eventName] = [callback];
+    }
+  }
+  emit(eventName) {
+    if (this.events[eventName]) {
+        this.events[eventName].forEach( callback => {
+        callback(eventName);
+      });
+    }
+  }
+  off(eventName, callback) {
+    let index = this.events[eventName].indexOf(callback);
+    delete this.events[eventName][index];
+  }
+}
+class Movie extends EventEmitter {
+  constructor(title, year, duration) {
+    super();
+    this.title = title;
+    this.year = year;
+    this.duration = duration;
+    this.cast = [];
+  }
+  play() {
+    this.emit('play');
+  }
+  pause() {
+    this.emit('pause');
+  }
+  resume() {
+    this.emit('resume');
+  }
+  addCast(actor) {
+    if (typeof actor === 'object' && actor.length) {
+      for (let ii = 0; ii < actor.length; ii++) {
```

Comment:
Now i get it... Anyway STOP USING THIS VARIABLES THAT ARE NOT UNDERSTANDABLE. Try to use variables that describes what you are doing